geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/SimpleFeatureConverterFactory.scala

```diff
@@ -348,13 +349,24 @@ trait ToSimpleFeatureConverter[I] extends SimpleFeatureConverter[I] with LazyLog
       i += 1
     }
 
-    val id = idBuilder.eval(t)(ec).asInstanceOf[String]
-    val sf = new ScalaSimpleFeature(targetSFT, id, sfValues)
-    userDataBuilder.foreach { case (k, v) => sf.getUserData.put(k, v.eval(t)(ec).asInstanceOf[AnyRef]) }
+    val sf = buildFeature(t, sfValues, ec)
 
     validate(sf, ec)
   }
 
+  private def buildFeature(t: Array[Any], sfValues: Array[AnyRef], ec: EvaluationContext): SimpleFeature = {
+    val sf = idBuilder match {
+      case LitNull => new ScalaSimpleFeature(targetSFT, null, sfValues)
```

Comment:
hmm, looking at the code, this will actually throw an NPE in `org.geotools.filter.identity.FeatureIdImpl`. so even though `id-field` is optional in the factory class, everything would fail if it's left out. we could just set this to an empty string here instead, I think that would work.